pynav/imm.py

```diff
@@ -0,0 +1,474 @@
+            Probability transition matrix corresponding to the IMM models.
+        """
+        self.kf_list = kf_list
+        self.Pi = Pi
+    
+    @staticmethod
+    def gaussian_mixing_vectorspace(weights: List[float], x_list: List[StateWithCovariance]):
+        """Calculate the mean and covariance of a Gaussian mixture on a vectorspace.
+
+        Parameters
+        ----------
+        weights : List[Float]
+            Weights corresponding to each Gaussian.
+        x_list : List[StateWithCovariance]
+            Gaussians to be mixed, with each element of x_list containing the mean
+            and covariance of the corresponding Gaussian. Each state must
+            be a VectorState.
+
+        Returns
+        -------
+        StateWithCovariance
+            A single StateWithCovariance containing the mean and covariance of the mixed Gaussian. 
+        """
+        x_bar = StateWithCovariance(
+                VectorState(np.zeros(x_list[0].state.value.shape), stamp=x_list[0].state.stamp),
+                np.zeros(x_list[0].covariance.shape)).copy()
+  
+        for (weight, x) in zip(weights, x_list):
+            x_bar.state.value = x_bar.state.value + weight*x.state.value 
+
+        for (weight, x) in zip(weights, x_list):
+            dx = (x.state.value-x_bar.state.value).reshape(-1,1)
+            x_bar.covariance = x_bar.covariance + weight*x.covariance + weight*dx @ dx.T
+
+        return x_bar
+
+    @staticmethod 
+    def reparametrize_gaussians_about_X_par(X_par: StateWithCovariance, X_list: List[StateWithCovariance]):
+        """Reparametrize each Lie group Gaussian in X_list about X_par. 
+        A Lie group Gaussian is only relevant in the tangent space of its own mean. 
+        To mix Lie group Gaussians, a specific X, X_par, must be chosen to expand 
+        a tangent space around. Once expanded in this common tangent space, 
+        the Gaussians may be mixed in a vector space fashion. 
+
+        Parameters
+        ----------
+        X_par : StateWithCovariance
+            Each member of X_list will be reparametrized as a Gaussian
+                in the tangent space of X_par. 
+        X_list : List[StateWithCovariance]
+            List of Lie group Gaussians to be reparametrized. 
+
+        Returns
+        -------
+        List[StateWithCovariance]
+            Where each state is a VectorState. These correspond to the mean and covariance
+            of X_list, expressed in the tangent space of X_par. 
+        """
+        x_reparametrized = []
+        g = X_par.group
+        if X_par.direction == "right":
+            for X in X_list: 
+                mu = g.Log(g.inverse(X_par.value) @ X.state.value)
+                Jr_inv = g.right_jacobian(mu)
+                Sigma = Jr_inv @ X.covariance @ Jr_inv.T 
+                mu = VectorState(mu, stamp = X_par.stamp)
+                x_reparametrized.append(StateWithCovariance(mu, Sigma))
+
+        return x_reparametrized
```

Comment:
```suggestion
       x_reparameterized = [] 
       for x in x_list:
           mu = x_par.minus(x)
           J = x_par.jacobian(mu)
           Sigma = J @ x.covariance @ J.T
           x_reparametrized.append(StateWithCovariance(mu, Sigma))
           
       x_reparameterized
```

should work for both vector space, liegroup (left/right), and any other state definition